alf/algorithms/algorithm.py

```diff
@@ -1200,16 +1248,17 @@ def _collect_train_info_sequentially(self, experience):
         return info
 
     def _collect_train_info_parallelly(self, experience):
-        batch_size = experience.step_type.shape[1]
-        length = experience.step_type.shape[0]
+        shape = alf.nest.get_nest_shape(experience)
+        length, batch_size = shape[:2]
 
         exp = alf.nest.map_structure(lambda x: x.reshape(-1, *x.shape[2:]),
                                      experience)
 
         if self._use_rollout_state:
             policy_state = exp.state
         else:
-            policy_state = self.get_initial_train_state(exp.step_type.shape[0])
+            size = alf.nest.get_nest_size(exp, dim=0)
+            policy_state = self.get_initial_train_state(size)
```

Comment:
Seems to me that the original code has a bug. Should use batch_size instead?

Reply:
This is after reshaping so now `size=length*batch_size` because of collecting info in parallel.